### ISSUE TITLE
tflite label_image not showing any labels

### ISSUE THREAD

**Issue by jbuisson1** [tensorflow/tensorflow#20485]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: OSX 10.13.3
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: v1.8.0-0-g93bc2e2072 1.8.0
- **Python version**: Python 3.6.0 :: Anaconda 4.3.1 (x86_64)
- **Bazel version (if compiling from source)**: Build label: 0.14.1-homebrew
- **GCC/Compiler version (if compiling from source)**: Apple LLVM version 9.1.0 (clang-902.0.39.2)
- **CUDA/cuDNN version**: NA
- **GPU model and memory**: NVIDIA GeForce GT 650M 1 GB
- **Exact command to reproduce**:  bazel-bin/tensorflow/contrib/lite/examples/label_image/label_image --tflite_model=../new_training_dir/retrainedMNetV1_7_8000.tflite --image=../tf_files/test_data_BITMAP/sidewalk/IMG_1913.bmp --labels=testLabels.txt

### Describe the problem
I have managed to compile the tflite label_image but when I use it with a retrained mobilenet I get the results without any labels next to it like following:

### Source code / logs
```
Loaded model ../new_training_dir/retrainedMNetV1_7_8000.tflite
resolved reporter
invoked
average time: 70.195 ms
8.15717: 544
6.8344: 732
6.73561: 734
6.2833: 423
5.93283: 420
```

**Comment by freedomtan** [comment]
interesting, does the labels.txt in this [zip](https://storage.googleapis.com/download.tensorflow.org/models/tflite/mobilenet_v1_224_android_quant_2017_11_08.zip) work for you?

**Comment by freedomtan** [comment]
@jbuisson1 probably your `testLabels.txt` is problematic. On macOS, it's supposed to be one line for each category (with lines ended with 0xa). See [the source code](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/examples/label_image/label_image.cc#L50-L72) for how labels are read.

**Comment by jbuisson1** [comment]
Here is the [testLabels.txt](https://github.com/tensorflow/tensorflow/files/2173832/testLabels.txt) file I am using, it doesn't seem to be much different from the other one except I only have 2 categories (the ones I retrained the mobilenet with).

```
cat -vet testLabels.txt
sidewalk$
crosswalk$
```
 
```
cat -vet labels.txt
background$
tench$
goldfish$
great white shark$
...
hen-of-the-woods$
bolete$
ear$
toilet tissue$
```

This labels file works with the label_image for the .pb file, it's also the one I use for TFLite on Android and the TFLiteClassifier works well with it, the only problem is that now the recognitions are not accurate at all compared to the ones made by the .pb file as it recognizes every image as the same category (I don't know if this might be related).

**Comment by freedomtan** [comment]
@jbuisson1 I don't get it. If you only have two categories, how come the results are showing that the highest one is category 544?

**Comment by jbuisson1** [comment]
Well I wasn't aware that was what these numbers meant, but your guess is as good as mine... I will upload my models asap if you want to take a look (maybe I did something wrong in the conversion to .tflite or something)

**Comment by jbuisson1** [comment]
I uploaded the models if it helps you see if anything is wrong on that end

**Comment by freedomtan** [comment]
It's my fault (I wrote the original label_image for tf lite). The number of outputs and number of results are fixed numbers now (see https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/examples/label_image/label_image.cc#L216-L237). Quick solution: changing output_size and num_results should fix the problem. 

I'll send a PR to add command options and check the output tensor later.

**Comment by jbuisson1** [comment]
It works with this quick fix! Thank you!!!

**Comment by karapapas** [comment]
> @jbuisson1: It's my fault (I wrote the original label_image for tf lite). The number of outputs and number of results are fixed numbers now (see https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/examples/label_image/label_image.cc#L216-L237). Quick solution: changing output_size and num_results should fix the problem.
> 
> I'll send a PR to add command options and check the output tensor later.
@freedomtan @jbuisson1 

I don't get it how exactly did you solve that? The link is dead, please give a hint.

**Comment by pradyu1** [comment]
The same issue after deploying the Tensorflow lite example on mobile with my custom classes it's not showing any labels.

**Comment by freedomtan** [comment]
@Karadarya this should be fixed by #20988 (a7fc381) already.
